pkg/render/intrusion_detection.go

```diff
@@ -372,6 +378,30 @@ func (c *intrusionDetectionComponent) intrusionDetectionClusterRole() *rbacv1.Cl
 		rules = append(rules, managementRule...)
 	}
 
+	if c.cfg.Tenant.MultiTenant() {
+		// These rules are used by Intrusion Detection Controller in a management cluster serving multiple tenants in order to appear to managed
+		// clusters as the expected serviceaccount. They're only needed when there are multiple tenants sharing the same
+		// management cluster.
+		rules = append(rules, []rbacv1.PolicyRule{
+			{
+				APIGroups:     []string{""},
+				Resources:     []string{"serviceaccounts"},
+				Verbs:         []string{"impersonate"},
+				ResourceNames: []string{IntrusionDetectionControllerName},
+			},
+			{
+				APIGroups: []string{""},
+				Resources: []string{"groups"},
+				Verbs:     []string{"impersonate"},
+				ResourceNames: []string{
+					serviceaccount.AllServiceAccountsGroup,
```

Comment:
Does it actually need to impersonate this? Can we just have `fmt.Sprintf("%s%s", serviceaccount.ServiceAccountGroupPrefix, IntrusionDetectionNamespace)` here?